torch/_meta_registrations.py

```diff
@@ -289,6 +289,16 @@ def meta_copy_(self, src, non_blocking=False):
     # which runs most of the meta checks that we care about.
     # In theory, we should make this more robust by carefully
     # auditing our C++ copy_() kernel and copying the checks here.
+
+    if self.numel() == 0:
```

Comment:
Factor this out to a `has_internal_overlap` helper ? Alternatively just directly invoke `torch._debug_has_internal_overlap` ?

also cc @bdhirsh, iirc remember you added a python impl of has_internal_overlap. link it ?

Reply:
I added a method `tensors_definitely_do_not_overlap(x, y)` ([here](https://github.com/pytorch/pytorch/blob/main/torch/_functorch/aot_autograd.py#L1654)).

I think using the existing C++ one bound to python is fine here, so that we can at least have the same amount of error checking compared to eager mode. Maybe at some point we can beef up the one I added to AOTAutograd so that it's usable here: the issue is that detecting overlap in all cases cannot be done in constant time, and so we either have to exhaustively check every spot in memory, or give up at some point.